### PR TITLE
Handle remaining refs to `RestApiVersion#V_7`

### PR DESCRIPTION
**Pull request by DaveCTurner**
Removes several more references to the now-unused `RestApiVersion#V_7`
constant, and decorates all remaining references with an `@UpdateForV9`
annotation so that they all have clear owners.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java [hunk 1]
```diff
@@ -354,20 +353,10 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
...
-        PARSER.declareInt(
-            ReindexRequest::setMaxDocsValidateIdentical,
-            new ParseField("max_docs").forRestApiVersion(RestApiVersion.onOrAfter(RestApiVersion.V_8))
-        );
         // avoid silently accepting an ignored size.
-        PARSER.declareInt(
-            (r, s) -> failOnSizeSpecified(),
-            new ParseField("size").forRestApiVersion(RestApiVersion.onOrAfter(RestApiVersion.V_8))
-        );
+        PARSER.declareInt((r, s) -> failOnSizeSpecified(), new ParseField("size"));
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java:359]: The parser should error on unknown from what I can tell. Maybe we should remove the `size` parse field altogether?

File: server/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java [hunk 2]
```diff
@@ -27,6 +28,7 @@ public class CommonTermsQueryBuilder extends AbstractQueryBuilder<CommonTermsQue
     public static final String COMMON_TERMS_QUERY_DEPRECATION_MSG = "Common Terms Query usage is not supported. "
         + "Use [match] query which can efficiently skip blocks of documents if the total number of hits is not tracked.";
 
+    @UpdateForV9(owner = UpdateForV9.Owner.SEARCH_FOUNDATIONS) // v7 REST API no longer exists: eliminate ref to RestApiVersion.V_7
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java:31]:
```suggestion
    @UpdateForV9(owner = UpdateForV9.Owner.SEARCH_RELEVANCE) // v7 REST API no longer exists: eliminate ref to RestApiVersion.V_7
```

File: server/src/main/java/org/elasticsearch/index/query/TypeQueryV7Builder.java [hunk 3]
```diff
@@ -26,6 +27,7 @@
 
 import java.io.IOException;
 
+@UpdateForV9(owner = UpdateForV9.Owner.SEARCH_FOUNDATIONS) // v7 REST API no longer exists: eliminate ref to RestApiVersion.V_7
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/index/query/TypeQueryV7Builder.java:30]:
```suggestion
@UpdateForV9(owner = UpdateForV9.Owner.SEARCH_RELEVANCE) // v7 REST API no longer exists: eliminate ref to RestApiVersion.V_7
```

**Review by benwtrent — COMMENTED** [review]
Two minor ownership comments & one comment about reindex. Other search related & ML things lgtm

File: server/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java (hunk 2 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java]: thanks, see 22ad65e4d49a768f11dba5ffc44549e46e446431.

File: server/src/main/java/org/elasticsearch/index/query/TypeQueryV7Builder.java (hunk 3 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/index/query/TypeQueryV7Builder.java]: thanks, see 22ad65e4d49a768f11dba5ffc44549e46e446431.

File: server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java:360]: Yeah not sure if we care about the specific message here or not - I'll let the distrib/indexing team decide that (see 458b0fb71153).

**Review by benwtrent — APPROVED** [review]
From Search side ++

**Review by jakelandis — APPROVED** [review]
LGTM

(w.r.t to the mechanical changes of removing dead code)

**Pull request merged**